gdpr/anonymizers/model_anonymizers.py

```diff
@@ -0,0 +1,148 @@
+        AnonymizedData(object=obj, field=name, expired_reason=legal_reason).save()
+
+    def get_anonymized_value_from_obj(self, field: FieldAnonymizer, obj: Model, name: str) -> Any:
+        """Get from field, obj and field name anonymized value."""
+        return field.get_anonymized_value_from_obj(obj, name)
+
+    def get_local_fields(self, fields: FieldMatrix) -> FieldList:
+        """Get Iterable of local fields from fields matrix."""
+        if fields == "__ALL__":
+            return self.fields.keys()
+        for i in fields:
+            if type(i) not in [str, list, tuple]:
+                raise ImproperlyConfigured()
+        local_fields = [i for i in fields if type(i) == str]
+        if "__ALL__" in local_fields:
+            return self.fields.keys()
+        return local_fields
+
+    def get_related_fields(self, fields: FieldMatrix) -> Dict[str, Any]:
+        """Get Dictionary of related fields from fields matrix."""
+        related_fields = [i for i in fields if type(i) in [list, tuple]]
+        for i in related_fields:
+            if len(i) != 2:
+                raise ImproperlyConfigured()
+        return {i[0]: i[1] for i in related_fields}
+
+    def parse_fields_matrix(self, fields: FieldMatrix) -> Tuple[FieldList, Dict[str, Any]]:
+        """Parse fields matrix into local fields and related fields"""
+        if type(fields) not in [str, tuple, list]:
+            raise ImproperlyConfigured()
+        return self.get_local_fields(fields), self.get_related_fields(fields)
+
+    def perform_anonymization(self, obj: Model, updated_data: dict,
+                              legal_reason: Optional[LegalReason] = None) -> None:
+        """Update data in database and mark them as anonymized."""
+        with transaction.atomic():
+            obj.__class__.objects.filter(pk=obj.pk).update(**updated_data)
+            for i in updated_data.keys():
+                self.mark_field_as_anonymized(obj, i, legal_reason)
+
+    def anonymize_qs(self, qs: QuerySet) -> None:
+        raise NotImplementedError()
+
+    def anonymize_obj(self, obj: Model, legal_reason: Optional[LegalReason] = None,
+                      purpose: Optional["AbstractPurpose"] = None,
+                      fields: FieldMatrix = "__ALL__"):
+        local_fields, related_fields = self.parse_fields_matrix(fields)
+
+        # Filter out already anonymized fields
+        local_fields_non_anonymized = [i for i in local_fields if not self.is_field_anonymized(obj, i)]
+        update_dict = {name: self.get_anonymized_value_from_obj(self[name], obj, name) for name in
+                       local_fields_non_anonymized}
+
+        self.perform_anonymization(obj, update_dict, legal_reason)
+
+        for key in related_fields.keys():
+            related_attribute = getattr(obj, key)
+            o_anonymizer = None
+            for o in related_attribute.all():
+                if not o_anonymizer:
```

Comment:
proč to nastavuješ až teď ane před cyklem?

Reply:
Protože se to zjišťuje z `o` ke kterému, když nemá žádné pod sebou tak ho nedokážu zjistit.

Ale tuhle část budu refaktorovat protože funguje jen pro reverzní vazbu z FK nebo M2M, ale @zzuzzy potřebuje. aby fungovala i pro FK a tudíž i pro OneToOne takže to musím opravit.